src/components/Typography.js

```diff
@@ -0,0 +1,43 @@
+import { createElement } from 'react';
+
+export const TypographyElements = {
+  "body1": "p",
+  "body-1": "p",
+  "body2": "p",
+  "body-2": "p",
+  "span": "span",
+  "h1small": "h1",
+  "h1-small": "h1",
+  "h1large": "h1",
+  "h1-large": "h1",
+  "h2": "h2",
+  "h3": "h3",
+}
+
+export default function Typography(
+  { elType, className = "", id, style, children }
+) {
+  if (!Object.keys(TypographyElements).includes(elType)) {
+    throw TypeError("Unsupported element type for Typography component")
+  };
+
+  if (elType === "body1" || elType === "body-1") {
```

Comment:
Lets also keep the scope limited by adding only 1 pattern of elType i.e. either "body1" or "body-1" to avoid a mixture of styles

Reply:
I suggest the hyphenated version, in that case, as it's easier to read.

Reply:
then lets keep the hyphenated version only

Reply:
after adding CSS modules, which recommend using camelCase, I went with unhypenated for all uses